File: record-generator/processor/src/main/java/me/bristermitten/mittenlib/codegen/record/RecordConstructorSpec.java
package me.bristermitten.mittenlib.codegen.record;

import com.squareup.javapoet.TypeName;

import java.util.List;

/**
 * A record with a single constructor
 *
 * @param name   the name of the record
 * @param fields the fields of the record
 */
public record RecordConstructorSpec(
        String name,
        List<RecordFieldSpec> fields
) {
    public record RecordFieldSpec(
            String name,
            TypeName type
    ) {
    }
}

File: record-generator/processor/src/main/scala/me/bristermitten/mittenlib/codegen/record/RecordConstructorSpec.scala
package me.bristermitten.mittenlib.codegen.record

import com.palantir.javapoet.TypeName

case class RecordConstructorSpec(
    name: String,
    fields: java.util.List[RecordConstructorSpec.RecordFieldSpec]
)

object RecordConstructorSpec:
  case class RecordFieldSpec(name: String, `type`: TypeName)
